src/harness/testcases/spectrum_inquiry_testcase.py

```diff
@@ -391,6 +392,87 @@ def test_WINNF_FT_S_SIQ_16(self):
       self.assertEqual(resp['response']['responseCode'], 0)
 
   @winnforum_testcase
+  def test_WINNF_FT_S_SIQ_18(self):
+    """This test ensures that when a message with two requests, 
+    one as a claimed PPA user and the other as a GAA user
+    Response Code must be 0 for all CBSDs
+    """
+
+    # Load the Data
+    pal_low_frequency = 3550000000.0
+    pal_high_frequency = 3560000000.0
+
+    device_a = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    pal_record = json.load(
+      open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+    ppa_record = json.load(
+      open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
+    ppa_record, pal_record = makePpaAndPalRecordsConsistent(ppa_record,
+                                                            [pal_record],
+                                                            pal_low_frequency,
+                                                            pal_high_frequency,
+                                                            device_a['userId'])
+    self._sas_admin.InjectZoneData({"record": ppa_record})
+    self._sas_admin.InjectPalDatabaseRecord(pal_record[0])
+    device_a['installationParam']['latitude'], device_a['installationParam']['longitude'] = \
+      getRandomLatLongInPolygon(ppa_record)
+    device_c = json.load(
+      open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    # Register the devices
+    request = {'registrationRequest': [device_a, device_c]}
+    response = self._sas.Registration(request)['registrationResponse']
+    # Check registration response
+    cbsd_ids = []
+    for resp in response:
+      self.assertEqual(resp['response']['responseCode'], 0)
+      cbsd_ids.append(resp['cbsdId'])
+    del request, response
+
+    # Inject Cluster List
+    cluster_list = {'zoneId': ppa_record['id'],
+                    'cbsdIds': [cbsd_ids[0]]}
+    self._sas_admin.InjectClusterList(cluster_list)
```

Comment:
I think you have missing two steps:
     1- Inject Pal Database Record
     2- Inject PPA record
bofore to inject cluster List
If you don't inject PPA in SAS, it will be not possible to associate CBSDs with PPA